backends/aerospike.go

```diff
@@ -100,21 +100,20 @@ func (a *AerospikeBackend) Put(ctx context.Context, key string, value string, tt
 	policy := &as.WritePolicy{Expiration: uint32(ttlSeconds)}
 
 	if err := a.client.Put(policy, asKey, bins); err != nil {
-		return formatAerospikeError(err, "PUT")
+		return formatAerospikeError(err)
 	}
 
 	return nil
 }
 
-func formatAerospikeError(err error, caller ...string) error {
+func formatAerospikeError(err error) error {
 	if err != nil {
-		msg := "Aerospike"
-		for _, str := range caller {
-			if len(str) > 0 {
-				msg = fmt.Sprintf("%s %s", msg, str)
+		if aerr, ok := err.(as_types.AerospikeError); ok {
+			if aerr.ResultCode() == as_types.KEY_NOT_FOUND_ERROR {
+				return KeyNotFoundError{"Aerospike"}
```

Comment:
Why not just log the error here with the desired level i.e. `log.Debug` for the `KEY_NOT_FOUND_ERROR` and `log.Error` for others and then return the final formatted error? This way you won't have to determine the log level and such later on. I also believe that would be much easier to read as well. Right now, you have to go back and forth and connect the dots to understand which error is getting logged at what log level

Reply:
We discussed and decided to merge with `determineLogLevel` with `TestLogAtLevel`.